.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: Java CI
+
+on: [push]
```

Comment:
can we also build on pull requests? maybe we skip the upload step for that though, but it lets us validate PRs as they come in, rather than merging first